packages/react/src/components/Storage/File-Uploader/FileUploader.tsx

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useEffect } from 'react';
+import { FileUploaderProps } from './types';
+import { UploaderButton } from './UploaderButton';
+import { Previewer } from './Previewer';
+import { View } from '../../../primitives';
+import { UploaderDrop } from './UploaderDrop';
+import { useFileUploader } from './hooks/useFileUploader';
+
+/**
+ * [📖 Docs](https://ui.docs.amplify.aws/react/connected-components/storage)
+ */
+
+export function FileUploader({
+  acceptedFileTypes,
+  fileNames,
+  isPreviewerVisible,
+  level,
+  components: customComponents = {},
+  maxFiles,
+  maxMultipleSize,
+  maxSize,
+  multiple = true,
+  onChange,
+  onError,
+  onSuccess,
+  path,
+  showPreview = false,
+  variation = 'button',
+}: FileUploaderProps): JSX.Element {
+  function Router(): JSX.Element {
```

Comment:
What is the purpose of `Router`? Also, this entire component is going to be recreated on every render, why can't we just return `Previewer` without this wrapper?

Reply:
I removed it in the next PR, but I'll remove it here too